### PR TITLE
Enable some basic configurations in a YAML file

### PR DESCRIPTION
**Pull request by dlmccoy**
**Why this PR?**
In the project I'd like to use this for, it's not feasible for me to use either the 'docs' folder or actually commit the generated YARD files to master. 

**What Changed?**
Add ability to include a YAML file which can configure the source directory and branch, as well as the destination branch.

### CODE REVIEW

File: lib/yard-ghpages.rb [hunk 1]
```diff
@@ -23,9 +24,20 @@ def install(opts)
 
           desc 'Publish documentation to gh-pages'
           task :publish do |t|
+            source_branch = 'master'
+            source_directory = 'docs'
+            destination_branch = 'gh-pages' 
+
+            config_filename = '.yard-gh-pages.yml'
```
Comment (trhodeos) [inline lib/yard-ghpages.rb:31]: I haven't used ruby or rake for a while, but does this work if someone calls rake from a subdirectory? IIRC rake will go to the parent directory until a Rakefile is found. I just want to make sure this always looks for the config file in the directory with the Rakefile, not the potential subdirectory that 'rake' is called from. Any idea if this works in that case?

**Review by trhodeos — COMMENTED** [review]
Nice, LGTM overall, just one minor test to run. Thanks for the contribution!

File: lib/yard-ghpages.rb (hunk 1 above)
Comment (dlmccoy) [reply lib/yard-ghpages.rb:31]: Yes, just tried this! When running from a subdirectory, it outputs what directory it's actually running in (in my case the root of the project)

Comment (trhodeos) [reply lib/yard-ghpages.rb:31]: Great, thanks!

**Pull request merged**